### PR TITLE
Patch in ecf04cc and 49f0fb9 from gRPC.

### PR DESCRIPTION
**Pull request by lberki**
This is so that Envoy keeps building with the --incompatible_disable_legacy_proto_provider command line option of Bazel which is to be flipped in 1.0 .

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -596,7 +596,14 @@ def _com_googlesource_quiche():
     )
 
 def _com_github_grpc_grpc():
-    _repository_impl("com_github_grpc_grpc")
+    _repository_impl(
+        "com_github_grpc_grpc",
+        patches = [
+            "@envoy//bazel:grpc-protoinfo-1.patch",
```
Comment (lizan) [inline bazel/repositories.bzl]: Can this two patch combined to one?

File: bazel/repositories.bzl [hunk 2]
```diff
@@ -596,7 +596,14 @@ def _com_googlesource_quiche():
     )
 
 def _com_github_grpc_grpc():
-    _repository_impl("com_github_grpc_grpc")
+    _repository_impl(
+        "com_github_grpc_grpc",
+        patches = [
```
Comment (lizan) [inline bazel/repositories.bzl]: Can you file an issue and mention it as a TODO comment here to remove the patches once a new gRPC is released? If the patch is based on a gRPC master PR, link there in the comment too.

**Comment by lizan** [comment]
Does this fix #7835 too?

File: bazel/repositories.bzl (hunk 1 above)
Comment (lberki) [reply bazel/repositories.bzl]: I certainly can, but are you sure that's what you want? This way, it each `.patch` file corresponds to a gRPC commit, which I thought was a nice property.

File: bazel/repositories.bzl (hunk 2 above)
Comment (lberki) [reply bazel/repositories.bzl]: https://github.com/envoyproxy/envoy/issues/7863 . Also added a comment.

**Comment by lberki** [comment]
Nope, those are different issues -- #7835 is about a breakage that stems from gRPC using properties of the output tree Bazel never guaranteed and is about breakage with Bazel 0.29, whereas #7863 is a known-incompatible change that will be broken in 1.0 .

File: bazel/repositories.bzl (hunk 1 above)
Comment (lizan) [reply bazel/repositories.bzl]: Ah I see, sure let’s keep as it is.

**Comment by lizan** [comment]
Can you add a patch for #7835 as well? That’s a breakage coming sooner.

**Comment by lberki** [comment]
Done. It's not merged yet, but I guess that's OK?

**Comment by lberki** [comment]
Paperwork complete!

**Comment by lberki** [comment]
Gentle ping! This fixes a breakage in the coming-very-soon Bazel 0.29.

**Comment by lizan** [comment]
@lberki we just had a security release today, I was holding this change and will merge later today.

**Comment by lberki** [comment]
Thank you! I verified that this fixes the potential breakage, so I'll submit our flag flip.

**Pull request merged**